examples/ui/borders.rs

```diff
@@ -89,25 +89,50 @@ fn setup(mut commands: Commands) {
                 style: Style {
                     width: Val::Px(10.),
                     height: Val::Px(10.),
+                    border_radius: UiBorderRadius::all(Val::Px(5.)),
                     ..Default::default()
                 },
                 background_color: Color::YELLOW.into(),
                 ..Default::default()
             })
             .id();
+        let border = borders[i % borders.len()];
+        let border_radius = UiBorderRadius::px(
+            if border.right != Val::Px(0.) && border.top != Val::Px(0.) {
+                f32::MAX
+            } else {
+                0.
+            },
+            if border.right != Val::Px(0.) && border.bottom != Val::Px(0.) {
+                f32::MAX
+            } else {
+                0.
+            },
+            if border.left != Val::Px(0.) && border.bottom != Val::Px(0.) {
+                f32::MAX
+            } else {
+                0.
+            },
+            if border.left != Val::Px(0.) && border.top != Val::Px(0.) {
+                f32::MAX
+            } else {
+                0.
+            },
+        );
```

Comment:
This is difficult to read. It discourages people from contributing.  More verbose is not necessarily equal to clearer.

```suggestion
        let border = borders[i % borders.len()];
        let non_zero = |val1, val2| val1 != Val::Px(0.) && val2 != Val::Px(0.);
        let border_size = |val1, val2| if non_zero(val1, val2) { f32::MAX } else { 0. };
        let border_radius = UiBorderRadius::px(
            border_size(border.top, border.right),
            border_size(border.bottom, border.right),
            border_size(border.bottom, border.left),
            border_size(border.top, border.left),
        );
```